File: src/ios/FamilyHub/Networking/AuthAPI.swift
import Foundation

extension APIClient {
    func login(_ request: LoginRequest) async throws -> AuthResponse {
        try await post("api/auth/login", body: request)
    }

    func register(_ request: RegisterRequest) async throws -> AuthResponse {
        try await post("api/auth/register", body: request)
    }
}

File: src/frontend/src/app/api/authApi.js
import { apiClient } from './client';
export async function register(request) {
    const response = await apiClient.post('/auth/register', request);
    return response.data;
}
export async function login(request) {
    const response = await apiClient.post('/auth/login', request);
    return response.data;
}
export async function changePassword(request) {
    const response = await apiClient.post('/auth/change-password', request);
    return response.data;
}
